### DISCUSSION TITLE
Trouble with text translation

### DISCUSSION THREAD

**Discussion by chianti-ga** [slint-ui/slint#3243] [Q&A] [answered]
Hi, I wanted to use the Slint translation system by following the documentation (https://slint.dev/releases/1.1.1/docs/slint/src/concepts/translations).

Unfortunately I can't get the strings translated, I also followed the "gallery" example (https://github.com/slint-ui/slint/blob/master/examples/gallery/) without success.

I checked to see if it was my code that was the problem, but to my great surprise, when I cloned the "gallery" exemple, the string translation didn't appear either.  

I'm verry confused, does anyone have anyclue on what i'm doing wrong? 
I appreciate any help, thanks in advance.

(I'm sorry for any mistakes or misunderstandings, English isn't my first language.)

Here's the projects files:
[project.zip](https://github.com/slint-ui/slint/files/12297674/project.zip)

**Comment by tronical** [comment]
Hi! Are you running your app on Linux? What do you see if you try to run `locale`, `LANG=fr_FR.utf-8 locale`, and `locale -a` on the command line?

If you don't see any translations then in my experience (after trying this) it's sometimes related to `setlocale()` failing, because it's missing the locale definitions.

**Reply by tronical** [reply, marked as answer]
I also tried your example now and it works for me on macOS if I set `LANG=fr`, with one caveat:

In your `src/main.rs` you initialise the base directory for your translations to be next to the executable:

```rust
slint::init_translations!(std::env::current_exe().unwrap().parent().unwrap().join("lang"));
```

If I run with `cargo run`, the executable will be in `target/debug/gmod_3d_packager` and consequently slint will try to look for translations `target/debug/lang` - which doesn't exist.

You can either copy the executable out of there into the root, or (for the purpose of development) use 

```rust
slint::init_translations!(concat!(env!("CARGO_MANIFEST_DIR"), "/lang/"));
```

**Reply by chianti-ga** [reply]
Thanks, it now works perfectly under linux but not under windows, any idea why?

**Reply by tronical** [reply]
Unfortunately, gettext on Windows is a little trickier and it doesn't work out of the box :( I've filed https://github.com/slint-ui/slint/issues/3307 with some details.
